state_base.py

```diff
@@ -49,8 +51,8 @@ def get_state(self, group):
         state.
         """
         if group not in self._states:
-            self._states[group] = copy(self._initial_state)
-        return self._states[group]
+            self._states_without_initial[group] = copy(self._initial_state)
```

Comment:
This variable name doesn't make sense. It's called `_states_without_initial` but this line is setting  some value you called `_initial_state` to it.